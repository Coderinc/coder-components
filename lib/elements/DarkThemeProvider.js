import React from 'react';
import { ThemeProvider } from 'styled-components';
import darkThemeBase from '../styles/themes/darkTheme';
import baseColors from '../styles/base-colors';

export default props => {
  const { primary = 'pink' } = props;

  const theme = {};
  Object.keys(darkThemeBase).forEach(k => {
    theme[k] = baseColors[darkThemeBase[k]] || darkThemeBase[k];
  });
  theme.primary = baseColors[primary];

  return (
    <ThemeProvider theme={theme}>
      {props.children}
    </ThemeProvider>
  );
};
