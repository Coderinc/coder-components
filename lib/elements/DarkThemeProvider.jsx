import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import darkThemeBase from '../styles/themes/darkTheme';
import baseColors from '../styles/base-colors';

const DarkThemeProvider = ({ primary, children }) => {
  const theme = {};
  Object.keys(darkThemeBase).forEach((k) => {
    theme[k] = baseColors[darkThemeBase[k]] || darkThemeBase[k];
  });
  theme.primary = baseColors[primary];

  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
};

DarkThemeProvider.propTypes = {
  primary: PropTypes.string,
  children: PropTypes.element,
};

DarkThemeProvider.defaultProps = {
  primary: 'pink',
  children: null,
};

export default DarkThemeProvider;
