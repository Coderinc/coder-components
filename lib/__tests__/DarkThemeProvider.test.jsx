import React from 'react';
import renderer from 'react-test-renderer';
import DarkThemeProvider from '../elements/DarkThemeProvider';

describe('DarkThemeProvider', () => {
  it('renders correctly', () => {
    const component = renderer.create(<DarkThemeProvider />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
