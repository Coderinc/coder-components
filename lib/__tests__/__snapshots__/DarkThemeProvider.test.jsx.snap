// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkThemeProvider renders correctly 1`] = `null`;
